samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.qnamaker;
+
+import com.microsoft.bot.ai.qna.QnAMaker;
+import com.microsoft.bot.ai.qna.QnAMakerEndpoint;
+import com.microsoft.bot.ai.qna.QnAMakerOptions;
+import com.microsoft.bot.builder.ActivityHandler;
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.integration.Configuration;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.CompletableFuture;
+
+public class QnABot extends ActivityHandler {
+    private static Configuration configuration;
+
+    public QnABot(Configuration withConfiguration) {
+        configuration = withConfiguration;
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        QnAMakerEndpoint qnAMakerEndpoint = new QnAMakerEndpoint();
+        qnAMakerEndpoint.setKnowledgeBaseId(configuration.getProperty("QnAKnowledgebaseId"));
+        qnAMakerEndpoint.setEndpointKey(configuration.getProperty("QnAEndpointKey"));
+        qnAMakerEndpoint.setHost(configuration.getProperty("QnAEndpointHostName"));
```

Comment:
If i'm not wrong, this can be reduced as:
```suggestion
        QnAMakerEndpoint qnAMakerEndpoint = new QnAMakerEndpoint() {{
            setKnowledgeBaseId(configuration.getProperty("QnAKnowledgebaseId"));
            setEndpointKey(configuration.getProperty("QnAEndpointKey"));
            setHost(configuration.getProperty("QnAEndpointHostName"));
        }};        
```

Reply:
excellent. Didn't know this syntax